modules/terraform-azure-nfs/main.tf

```diff
@@ -0,0 +1,142 @@
+  resource_group_name = var.resourcegroup_name
+  location = var.azure_region
+  account_tier = "Premium"
+  account_replication_type = "LRS"
+  account_kind = "FileStorage"
+  enable_https_traffic_only = false
+}
+
+
+resource "azurerm_storage_share" "nfs_storage_share" {
+  name = var.nfs_file_share_name
+  storage_account_name = azurerm_storage_account.nfs_storage_account.name
+  enabled_protocol = "NFS"
+  quota = var.nfs_file_share_size
+}
+
+
+resource "azurerm_private_dns_zone" "nfs_privatednszone" {
+  name = "privatelink.file.core.windows.net"
+  resource_group_name = var.resourcegroup_name
+}
+
+resource "azurerm_private_dns_zone_virtual_network_link" "nfs_vnet_link" {
+  name = "${var.env_prefix}_vnetlink"
+  resource_group_name = var.resourcegroup_name
+  private_dns_zone_name = azurerm_private_dns_zone.nfs_privatednszone.name
+  virtual_network_id = data.azurerm_virtual_network.nfs_vnet.id
+}
+
+
+resource "azurerm_private_endpoint" "nfs_private_endpoint" {
+  name = "${var.env_prefix}_${data.azurerm_subnet.nfs_subnet.name}_nfs_private_endpoint"
+  location = var.azure_region
+  resource_group_name = var.resourcegroup_name
+  subnet_id = data.azurerm_subnet.nfs_subnet.id
+
+  private_service_connection {
+    name = "nfs-privateserviceconnection"
+    private_connection_resource_id = azurerm_storage_account.nfs_storage_account.id
+    subresource_names = [
+
+      "file",
+    ]
+    is_manual_connection = false
+  }
+
+  private_dns_zone_group {
+    name = "nfs-dns-zone-group"
+    private_dns_zone_ids = [
+      azurerm_private_dns_zone.nfs_privatednszone.id]
+  }
+}
+
+resource "azurerm_public_ip" "nfsvm_public_ip" {
+  name = "${var.env_prefix}nfsvm-publicip"
+  resource_group_name = var.resourcegroup_name
+  location = var.azure_region
+  allocation_method = "Static"
+  sku = "Standard"
+}
+
+resource "azurerm_network_interface" "nfsvm_nic" {
+  name = "${var.env_prefix}nfsvm-nic"
+  resource_group_name = var.resourcegroup_name
+  location = var.azure_region
+
+  ip_configuration {
+    name = "internal"
+    subnet_id = data.azurerm_subnet.nfs_subnet.id
+    private_ip_address_allocation = "Dynamic"
+    public_ip_address_id = azurerm_public_ip.nfsvm_public_ip.id
+  }
+}
+
+resource "azurerm_network_security_group" "nfsvm_sg" {
+  name = "${var.env_prefix}nfsvm-sg"
+  resource_group_name = var.resourcegroup_name
+  location = var.azure_region
+
+  security_rule {
+    name = "allowssh"
+    priority = 100
+    direction = "Inbound"
+    access = "Allow"
+    protocol = "Tcp"
+    source_port_range = "*"
+    destination_port_range = "22"
+    source_address_prefix = "*"
+    destination_address_prefix = "*"
+  }
+}
+
+resource "azurerm_network_interface_security_group_association" "nfsvm_nic_sg" {
+  network_interface_id = azurerm_network_interface.nfsvm_nic.id
+  network_security_group_id = azurerm_network_security_group.nfsvm_sg.id
+}
+
+
+resource "azurerm_linux_virtual_machine" "nfs_vm" {
```

Comment:
Suggest to create a create_vm bool input variable to control the creation of the VM.

Use the variable with e.g. `count = var.create_vm ? 1 : 0`

Reply:
will think about this later, didn't add it now because we need VM creation always.

Reply:
@tush4hworks - I would like to incorporate the variable, even with default value of true.  We can work together on this to get it into this PR. 

As the VM is potentially short-lived and only used for a single purpose of performing a `chown` on the nfs directory it should be optional.